### DISCUSSION TITLE
Community PRs to be merged in Oct 2022

### DISCUSSION THREAD

**Discussion by gurpreet-wpf** [dotnet/wpf#7130] [Announcements]
Hi All,

Below PRs will be merged in upcoming month (Oct '22) after the DRTs and Microsuite Test Community Pass result:

1. #7091 
2. #7015 
3. #7000 
4. #6889 
5. #6862 
6. #6857 
7. #6785 
8. #6779 
9. #6714 
10. #5066 
11. #3008 

Thanks!

**Comment by kant2002** [comment]
Will DRT pass would happen on private infra, or it would happen as part of existing CI infra?

**Reply by singhashish-wpf** [reply]
Limited number of DRTs which are open sourced in wpf-test repo will be run as a part of existing CI infra.
However not just DRTs are run on these PRs, these will undergo full test pass and that would happen on private internal infra.

**Reply by kant2002** [reply]
My concern is how much time passed between time develop submit PR and time when PR lands after all checks. Right now, I think this is about 1 month or two which kills any incentive to contribute. Anyway, thank you for continuing this unpleasant discussion with me.

**Reply by pchaurasia14** [reply]
@kant2002 - We acknowledge the problem and we are working actively to address it. Over last few months, we've increased the number of CTP runs that we do in a month. Due to some unfortunate circumstances, we had to reduce the frequency of CTP in the last month to accommodate for few other build/release related tasks. However, we're almost done with those tasks and now can resume the CTP runs back again. 

We're hopeful to get things back on track and will reduce the total turnaround times for PRs. 
Thanks for your patience.

**Comment by wstaelens** [comment]
what is the status of https://github.com/dotnet/wpf/discussions/6556#discussioncomment-2722928 ???

**Reply by lindexi** [reply]
@wstaelens  Seems I fix one of the XPS bug in https://github.com/dotnet/wpf/pull/7091

**Reply by wstaelens** [reply]
@lindexi ok great! One down, multiple to go.
Can you have a look at this one: https://github.com/dotnet/wpf/issues/3546#issuecomment-1166977050 
You will see that KevinCathcart wrote a proof of concept to fix it but needs to be added and tested: https://github.com/dotnet/runtime/compare/main...KevinCathcart:interleave

**Comment by pchaurasia14** [comment]
Of all the above PRs that were scheduled to merged for Oct CTP, we ran into test issues with few of them. 

- #6862
- #6779
- #3008
- #7015 

We're currently working to address those failures.

**Reply by wstaelens** [reply]
@pchaurasia14 ok, but please look at https://github.com/dotnet/wpf/issues/3546#issuecomment-1166977050
Open since first release of .net core

**Comment by kant2002** [comment]
What can be done to speed up PR landing main? WPF as was announced is community project. But unfortunately community can only write code, but never langing it still in the hands of Microsoft. Which is fine, but we still as community experience large delays between time when PR was filed and when it's landed. At the same time, MS contribution still flowing.

I think once all community contributions lands, it would be much easier to wait for new PR review process. Right now things are unnescessary slow for community project.

**Reply by pchaurasia14** [reply]
We've realized that the term "community run projects" is a broad term and is open to different interpretations by developers. However, the intended meaning of this term was to have a channel through which we can prioritize issues/PRs that the community thinks would improve WPF over other existing open PRs/issues (https://github.com/orgs/dotnet/projects/146).

In future, we'll refer to such initiatives by a different name (TBD -  _To Be Decided_) which may be less ambiguous.

Now, coming to what can be done to speed up PR landing process. Our primary goal is to ensure reduced turn-around-times for new PRs as well as clear the pending backlog of existing PRs. To that end, we are working on putting appropriate automation in place to reduce the dependency on manual testing as much as we can. 

Considering the number of community PRs that were merged in 2022 (which is higher as compared to previous years), we intend to be more consistent in maintaining a healthy rate of manual reviews going forward.

As for your observation on MS contribution still flowing, those are part of automated build coherence with other dotnet repositories. So, they do not have to go through manual scrutiny over behavioral changes.

**Reply by wstaelens** [reply]
"reduced ... for new PR"

A new name TBD will not change this bad running machine. Don't waste time on such things. Rather fix old bugs and PR's
